netmiko/nokia/nokia_srl.py

```diff
@@ -0,0 +1,197 @@
+    A:ams01#
+
+    --{ * candidate private private-admin }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    --{ + candidate private private-admin }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    This class support the default prompt configuration.
+    """
+
+    def session_preparation(self) -> None:
+        self._test_channel_read()
+        self.ansi_escape_codes = True
+        # Bottom toolbar text not required
+        command = "environment cli-engine type basic"
+        self.write_channel(self.normalize_cmd(command))
+        command = "environment complete-on-space false"
+        self.write_channel(self.normalize_cmd(command))
+        self.set_base_prompt()
+        time.sleep(10 * self.global_delay_factor)
+        self.clear_buffer()
+
+    def set_base_prompt(
+        self,
+        pri_prompt_terminator: str = "#",
+        alt_prompt_terminator: str = "",
+        delay_factor: float = 1.0,
+        pattern: Optional[str] = r"(\n\-\-\{.+\}\-\-\[.*\]\-\-\n.*# )",
+    ) -> str:
+        return super().set_base_prompt(
+            pri_prompt_terminator, alt_prompt_terminator, delay_factor, pattern
+        )
+
+    def config_mode(
+        self,
+        config_command: str = "enter candidate private",
+        pattern: str = "]--",
+        re_flags: int = 0,
+    ) -> str:
+
+        output = super().config_mode(
+            config_command=config_command, pattern="", re_flags=re_flags
+        )
+        return output
+
+    def check_config_mode(
+        self,
+        check_string: str = r"\n--{( | \* | \+ | \+\* | \!\+ | \!\* )candidate",
+        pattern: str = " ]--",
+    ) -> bool:
+        self.write_channel(self.RETURN)
+        """
+        supported first line prompt configuration:
+        {modified_flags}{mode_and_session} }}--[ {pwc} ]
+        """
+        if not pattern:
+            output = self.read_channel_timing(read_timeout=10.0)
+        else:
+            output = self.read_until_pattern(pattern=pattern)
+
+        matches = re.search(check_string, output)
+        return True if matches else False
+
+    def commit(self) -> str:
+        """Commit changes by using 'commit stay'."""
+        cmd = "commit stay"
+        self.write_channel(self.normalize_cmd(cmd))
+        output = (
+            self._get_cmd_output_and_prompt(cmd)
+            if self.global_cmd_verify is not False
+            else ""
+        )
+        return output
+
+    def save_config(
+        self,
+        cmd: str = "save startup",
+        confirm: bool = False,
+        confirm_response: str = "",
+    ) -> str:
+        """Save current running configuration as initial (startup) configuration"""
+        self.write_channel(self.normalize_cmd(cmd))
+        output = (
+            self._get_cmd_output_and_prompt(cmd)
+            if self.global_cmd_verify is not False
+            else ""
+        )
+        return output
```

Comment:
Why the need of this private method?

I would have a pretty big bias to doing the following instead:

```python
return self._send_command_str(cmd)
```

I say this as I will forget about this private method and then accidentally break your driver. In other words, it is better to follow more standard Netmiko practices unless we can't make it work (and I am not seeing here why the standard practices wouldn't work).